Sources/SwiftParserDiagnostics/MissingNodesError.swift

```diff
@@ -290,7 +290,68 @@ public struct InsertTokenFixIt: ParserFixIt {
 // MARK: - Generate Error
 
 extension ParseDiagnosticsGenerator {
-  func handleMissingSyntax<T: SyntaxProtocol>(_ node: T) -> SyntaxVisitorContinueKind {
+  func handleMissingToken(_ node: TokenSyntax) {
+    guard let invalidToken = node.previousToken(viewMode: .all),
+          let unexpectedTokens = invalidToken.parent?.as(UnexpectedNodesSyntax.self),
+          unexpectedTokens.count == 1 else {
+      _ = handleMissingSyntax(node)
+      return
+    }
+
+    // The previous token is unexpected, assume that it was intended to be
+    // this token.
+
+    if node.tokenKind.isIdentifier {
+      let fixIts: [FixIt]
+      if invalidToken.tokenKind.isKeyword || invalidToken.tokenKind.isDollarIdentifier {
+        // TODO: Should the parser add the text with backticks to the missing
+        // node? Then this could just make missing/present.
+        fixIts = [FixIt(message: .wrapInBackticks, changes: [
+          .replace(
+            oldNode: Syntax(invalidToken),
+            newNode: Syntax(TokenSyntax.identifier("`\(invalidToken.text)`", leadingTrivia: invalidToken.leadingTrivia, trailingTrivia: invalidToken.trailingTrivia))
+          )
+        ])]
+      } else {
+        fixIts = []
+      }
+      addDiagnostic(
+        invalidToken,
+        InvalidIdentifierError(invalidIdentifier: invalidToken, missingIdentifier: node),
+        fixIts: fixIts,
+        handledNodes: [unexpectedTokens.id]
+      )
+    } else if node.tokenKind == .period && invalidToken.tokenKind == .period {
+      // Trailing trivia is the cause of this diagnostic, don't transfer it.
+      let changes: [FixIt.Changes] = [
+        .makeMissing(invalidToken, transferTrivia: false),
+        .makePresent(node)
+      ]
+
+      if let identifier = node.nextToken(viewMode: .all),
+         identifier.rawTokenKind == .identifier,
+         identifier.presence == .missing {
+        // The extraneous whitespace caused a missing identifier, output a
+        // diagnostic for it instead. Insert non-whitespace trivia after the
+        // missing identifier.
```

Comment:
It took me a while to understand what “instead” was referring to. 

```suggestion
        // The extraneous whitespace caused a missing identifier, output a
        // diagnostic inserting it instead of a diagnostic to fix the trivia around the period. Insert non-whitespace trivia after the
        // missing identifier.
```